clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TargetHealthCachingAgent.java

```diff
@@ -0,0 +1,174 @@
+import com.amazonaws.services.elasticloadbalancingv2.model.DescribeTargetHealthRequest;
+import com.amazonaws.services.elasticloadbalancingv2.model.DescribeTargetHealthResult;
+import com.amazonaws.services.elasticloadbalancingv2.model.TargetHealthDescription;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.cats.agent.*;
+import com.netflix.spinnaker.cats.cache.*;
+import com.netflix.spinnaker.cats.provider.ProviderCache;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials;
+import com.netflix.spinnaker.clouddriver.core.provider.agent.HealthProvidingCachingAgent;
+import com.netflix.spinnaker.clouddriver.ecs.cache.Keys;
+import com.netflix.spinnaker.clouddriver.ecs.cache.client.EcsTargetGroupCacheClient;
+import com.netflix.spinnaker.clouddriver.ecs.cache.model.EcsTargetHealth;
+import com.netflix.spinnaker.clouddriver.ecs.model.loadbalancer.EcsTargetGroup;
+import com.netflix.spinnaker.clouddriver.ecs.provider.EcsProvider;
+import java.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TargetHealthCachingAgent extends AbstractEcsAwsAwareCachingAgent<EcsTargetHealth>
+    implements HealthProvidingCachingAgent {
+
+  private static final Collection<AgentDataType> types =
+      Collections.unmodifiableCollection(Arrays.asList(AUTHORITATIVE.forType(TARGET_HEALTHS.ns)));
+  private static final String HEALTH_ID = "ecs-alb-target-health";
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+  private final ObjectMapper objectMapper;
+
+  public TargetHealthCachingAgent(
+      NetflixAmazonCredentials account,
+      String region,
+      AmazonClientProvider amazonClientProvider,
+      AWSCredentialsProvider awsCredentialsProvider,
+      ObjectMapper objectMapper) {
+    super(account, region, amazonClientProvider, awsCredentialsProvider);
+    this.objectMapper = objectMapper;
+  }
+
+  @Override
+  protected List<EcsTargetHealth> getItems(AmazonECS ecs, ProviderCache providerCache) {
+    if (awsProviderCache == null) {
+      throw new NullPointerException("awsProviderCache not initialized on " + getAgentType() + ".");
+    }
+
+    EcsTargetGroupCacheClient targetGroupCacheClient =
+        new EcsTargetGroupCacheClient(awsProviderCache, objectMapper);
+    Set<String> targetGroups = fetchTargetGroups(targetGroupCacheClient);
+
+    AmazonElasticLoadBalancing amazonLoadBalancing =
+        amazonClientProvider.getAmazonElasticLoadBalancingV2(account, region, false);
+
+    List<EcsTargetHealth> targetHealthList = new LinkedList<>();
+
+    if (targetGroups != null) {
+      for (String tgArn : targetGroups) {
+        DescribeTargetHealthResult describeTargetHealthResult =
+            amazonLoadBalancing.describeTargetHealth(
+                new DescribeTargetHealthRequest().withTargetGroupArn(tgArn));
```

Comment:
Need to handle TargetGroupNotFoundException in case the target group cache is stale